GUI/Forms/ExtractProgressForm.cs

```diff
@@ -49,60 +47,65 @@ public ExtractProgressForm(Package package, TreeNode root, string path, ExportDa
             filesToExtract = new Queue<PackageEntry>();
             extractedFiles = new HashSet<string>();
 
-            this.package = package;
-            this.root = root;
             this.path = path;
+            this.decompile = decompile;
             this.exportData = exportData;
 
-            if (Decompile)
+            if (decompile)
             {
                 gltfExporter = new GltfModelExporter()
                 {
                     FileLoader = exportData.VrfGuiContext.FileLoader,
+                    ProgressReporter = new Progress<string>(SetProgress),
                 };
             }
         }
 
         protected override void OnShown(EventArgs e)
         {
+            if (ShownCallback != null)
+            {
+                ShownCallback(this);
+                return;
+            }
+
             Task
                 .Run(
                 async () =>
                 {
-                    Invoke((Action)(() =>
-                    {
-                        extractStatusLabel.Text = "Calculating...";
-                        extractProgressBar.Style = ProgressBarStyle.Marquee;
-                    }));
-
                     Console.WriteLine($"Folder export started to \"{path}\"");
-                    CalculateFilesToExtract(root);
 
-                    Invoke((Action)(() =>
+                    Invoke(() =>
                     {
+                        extractStatusLabel.Text = "Calculating...";
                         extractProgressBar.Style = ProgressBarStyle.Continuous;
-                    }));
+                    });
 
-                    if (Decompile)
+                    if (decompile)
                     {
                         foreach (var resourceType in ExtractOrder)
                         {
-                            Invoke(() => Text = $"Extracting {resourceType}s...");
+                            Invoke(() => extractStatusLabel.Text = $"Extracting {resourceType}s...");
                             var extension = FileExtract.GetExtension(resourceType);
                             await ExtractFilesAsync(filesToExtractSorted[extension + "_c"]).ConfigureAwait(false);
                         }
 
-                        Invoke(() => Text = $"Extracting files...");
+                        Invoke(() => extractStatusLabel.Text = $"Extracting files...");
```

Comment:
The subsequent `ExtractFilesAsync` immediately replaces these messages. Should keep them on title. When running a materials folder the title should say "Extracting Materials...", then "Extracting files..." when it moves to anything non-vmat. Also the "Calculating..." above doesn't seem to calculate anymore.

Reply:
I changed it to a log window.